File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/PersonDao.java
package com.charlezz.javaapp.feature.local;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface PersonDao {
    @Query("SELECT * FROM person")
    DataSource.Factory<Integer, Person> getPersonSource();
}

File: app/src/main/java/com/charlezz/finalarchitecture/feature/local/PersonDao.kt
package com.charlezz.finalarchitecture.feature.local

import android.arch.paging.DataSource
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Query


@Dao
interface PersonDao {
    @Query("SELECT * FROM person")
    fun getPersonSource():DataSource.Factory<Int, Person>
}